pgjdbc/src/main/java/org/postgresql/util/ByteConverter.java

```diff
@@ -65,6 +65,7 @@ private static void digitToString(int idx, short[] digits, CharBuffer buffer, bo
       boolean putit = (d1 > 0);
       if (putit || alwaysPutIt) {
         buffer.put((char)(d1 + '0'));
+        alwaysPutIt = true;
```

Comment:
This logic is rather confusing. Once we set alwaysPutIt to true it stays that way. Seems convoluted.

Reply:
Well, the better approach is probably to precompute digits, however, this seems to be the minimal change that fixes the issue.

Indeed, if we have started printing the number, then we always want to put subsequent digits even in case they are 0. That is why `alwaysPutIt = true` works here.

Reply:
Comments would suffice...